src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -24,7 +24,7 @@ The .NET Foundation licenses this file to you under the MIT license.
     <LinkerFlavor Condition="'$(LinkerFlavor)' == '' and '$(_linuxLibcFlavor)' == 'bionic'">lld</LinkerFlavor>
     <LinkerFlavor Condition="'$(LinkerFlavor)' == '' and '$(_targetOS)' == 'linux'">bfd</LinkerFlavor>
     <IlcDefaultStackSize Condition="'$(IlcDefaultStackSize)' == '' and '$(_linuxLibcFlavor)' == 'musl'">1572864</IlcDefaultStackSize>
-    <UseSystemZlib Condition="!Exists('$(IlcSdkPath)libz.a')">true</UseSystemZlib>
+    <UseSystemZlib Condition="'$(UseSystemZlib)' == '' and !Exists('$(IlcFrameworkNativePath)libz.a')">true</UseSystemZlib>
```

Comment:
Nit: I do not think that adding `"'$(UseSystemZlib)' == ''` condition here enables any new interesting scenarios. Is this trying to add support zlib that is neither the system zlib nor the system one? I think more changes would be needed for that.

Reply:
However, this change should be harmless, so it can stay as part this PR.

Reply:
If user had option for both, they can chose to opt for system zlib.

Reply:
I noticed that this targets file gets executed several times, but I am unsure if the UseSystemZlib property would preserve the value for each project executed. So it made sense to me to use this suggestion so that we didn't call Exists more times than necessary. I can revert it if it's not really needed.

Reply:
This targets file may get executed multiple times during dotnet/runtime build, but it should not ever be in a context shared by multiple projects. I do not think you are saving any Exists execution by adding this condition.

But as I have said, this change should be harmless and it matches common pattern used by properties, so it can stay in this PR.